http-client.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+type Proxy struct {
+	URL      string
+	Username string
+	Password string
+}
+
+func (p Proxy) HasAuthorizationCredentials() bool {
+	return p.Username != "" && p.Password != ""
```

Comment:
IS this line correct?
A proxy authentication can require username without password(i.e. password is "").

Reply:
That is true. I will correct this.